relay/blockchain/delegate.py

```diff
@@ -64,7 +65,9 @@ def calc_next_nonce(self, identity_address: str):
         except UnexpectedIdentityContractException:
             raise InvalidIdentityContractException
 
-    def calculate_fees_for_meta_transaction(self, meta_transaction: MetaTransaction):
+    def calculate_fees_for_meta_transaction(
+        self, meta_transaction: MetaTransaction
+    ) -> DelegationFees:
         try:
             valid = self.delegate.validate_nonce(meta_transaction)
```

Comment:
why do we validate the nonce here?

Reply:
That's the only thing we can validate and it should allow failing earlier if the nonce is wrong.

Reply:
yes, I think it is just a weird place to do that.